daemon/container.go

```diff
@@ -98,8 +98,6 @@ func (daemon *Daemon) Register(c *container.Container) error {
 
 	daemon.containers.Add(c.ID, c)
 	daemon.idIndex.Add(c.ID)
```

Comment:
Looking at these, it appears that we're ignoring any error that can occur when adding to the index; https://github.com/docker/docker/blob/fe9f5610c49b1a6ca2ebbb68c1478fc7f106eaa8/pkg/truncindex/truncindex.go#L59-L81

wondering if we should actually return those errors instead

/cc @cpuguy83